op-supervisor/supervisor/backend/db/sync/client.go

```diff
@@ -0,0 +1,226 @@
+	return nil
+}
+
+// SyncDatabase downloads the named file from the server.
+// If the local file exists, it will attempt to resume the download if resume is true.
+func (c *Client) SyncDatabase(ctx context.Context, chainID types.ChainID, database Database, resume bool) error {
+	// Validate file alias
+	filePath, exists := Databases[database]
+	if !exists {
+		return fmt.Errorf("unknown file alias: %s", database)
+	}
+
+	// Ensure the chain directory exists
+	chainDir := filepath.Join(c.config.DataDir, chainID.String())
+	if err := os.MkdirAll(chainDir, 0755); err != nil {
+		return fmt.Errorf("failed to create chain directory: %w", err)
+	}
+
+	filePath = filepath.Join(chainDir, filePath)
+
+	// Ensure the database file exists and get initial size
+	filePath = filepath.Join(chainDir, filePath)
+	var initialSize int64
+	if stat, err := os.Stat(filePath); err == nil {
+		initialSize = stat.Size()
+	}
+
+	// If we have data already and don't want to resume then stop now
+	if initialSize > 0 && !resume {
+		return nil
+	}
+
+	// Keep track of the current retry delay
+	currentDelay := c.retryDelay
+	for {
+		// Try to sync and return if successful
+		err := c.attemptSync(ctx, chainID, database, filePath, initialSize)
+		if err == nil {
+			return nil
+		}
+
+		// Check if context was canceled
+		if ctx.Err() != nil {
+			return ctx.Err()
+		}
+		c.logError("sync attempt failed", err, database)
+
+		// Wait before retrying
+		select {
+		case <-ctx.Done():
+			return ctx.Err()
+		case <-time.After(currentDelay):
+			// Double the delay for next time, up to max
+			currentDelay *= 2
+			if currentDelay > MaxRetryDelay {
+				currentDelay = MaxRetryDelay
+			}
+		}
+	}
+}
+
+// attemptSync makes a single attempt to sync the file
+func (c *Client) attemptSync(ctx context.Context, chainID types.ChainID, database Database, absPath string, initialSize int64) error {
+	// First do a HEAD request to get the file size
+	req, err := http.NewRequestWithContext(ctx, http.MethodHead, c.buildURL(chainID, database), nil)
+	if err != nil {
+		return fmt.Errorf("failed to create HEAD request: %w", err)
+	}
+	resp, err := c.httpClient.Do(req)
+	if err != nil {
+		return fmt.Errorf("HEAD request failed: %w", err)
+	}
+	if err := resp.Body.Close(); err != nil {
+		return err
+	}
+	if resp.StatusCode != http.StatusOK {
+		return fmt.Errorf("HEAD request failed with status %d", resp.StatusCode)
+	}
+	totalSize, err := parseContentLength(resp.Header)
+	if err != nil {
+		return fmt.Errorf("invalid Content-Length: %w", err)
+	}
+
+	// If we already have the whole file, we're done
+	if initialSize == totalSize {
+		return nil
+	}
+
+	// Create the GET request
+	req, err = http.NewRequestWithContext(ctx, http.MethodGet, c.buildURL(chainID, database), nil)
+	if err != nil {
+		return fmt.Errorf("failed to create GET request: %w", err)
+	}
+
+	// If we have partial file, try to resume
+	if initialSize > 0 {
+		req.Header.Set("Range", fmt.Sprintf("bytes=%d-", initialSize))
+	}
```

Comment:
I think you could expand the `op-service.BasicHTTPClient` to support partial Gets, instead of shimming it here. It'd be nice functionality to add for all

(but if it's actually some trouble to add it because this use case is very specific, I would concede this quickly just lmk)

Reply:
This seems to be already supported I just wasn't aware of the BasicHTTPClient. Updated to use it.